astropy/io/fits/_tiled_compression/codecs.py

```diff
@@ -334,7 +334,7 @@ def decode(self, buf):
         buf : np.ndarray
             The decompressed buffer.
         """
-        cbytes = np.frombuffer(buf, dtype=np.uint8).tobytes()
+        cbytes = np.frombuffer(_as_native_endian_array(buf), dtype=np.uint8).tobytes()
```

Comment:
Why only for this one and not for RICE or HCOMPRESS ?

Reply:
Good catch, thanks!